dandiapi/api/tasks/__init__.py

```diff
@@ -237,3 +237,10 @@ def _create_doi(version_id: int):
         version.save()
 
     transaction.on_commit(lambda: _create_doi(new_version.id))
+
+
+def _log_version_action(msg: str, version_id: int, version: Version) -> None:
+    """Centralize/harmonize logging of operations over a Version."""
+    logger.info(
+        '%s for version %d %s:%s', msg, version_id, version.dandiset.identifier, version.version
+    )
```

Comment:
> May be `str` of `version` has changed since when I submitted PR - I think I had checked that there was explicit need so I could see both DB level `id` (an int) and then information about dandiset version, such as 000026/draft or alike. Would both be visible with just `{version.id}, {version}`? (can't check ATM)

Correct, `{version.id}, {version}` will produce something like: `'123, 000026/draft'`.

> using f-strings in `.log` invocations IMHO should generally be discouraged since
> 
> * it causes f-formatting/stringification of all those values regardless of the active logging level (waste of cycles, despite being minor is a waste so why not to avoid it?)
> * if later decide to mix in adding proper `%-interpolations` with "args" for `.log` invocation, the string would need to be de-f-stringed since if any of the elements (msg, version or version.id) has any `%` (possibly not something guaranteed to not exist in those) it could cause entire %-interpolation to kaboom. So then could be forgotten to be done, missed, and the bomb would tick until something has `%` in it.


Seems like a very arbitrary concern, but I won't block on it.

Reply:
> - it causes f-formatting/stringification of all those values regardless of the active logging level (waste of cycles, despite being minor is a waste so why not to avoid it?)

Answering "why not avoid it?:" -- the readability of an f-string far outweighs any performance concerns due to using them.

> - if later decide to mix in adding proper %-interpolations with "args" for .log invocation, the string would need to be de-f-stringed since if any of the elements (msg, version or version.id) has any % (possibly not something guaranteed to not exist in those) it could cause entire %-interpolation to kaboom. So then could be forgotten to be done, missed, and the bomb would tick until something has % in it.

I'm not sure why you would need to convert back to the "args" style if we've adopted f-strings in usages like this. If it needs to be done, then we can always change it back but I'd argue to use f-strings here for readability reasons (and because I'm truly not concerned about "wasting cycles").

Reply:
To correct & elaborate on Yarik's second point: Even if you never decide to use `%`-formatting explicitly, the `logging` module will always try to interpolate `%` sequences in log messages, and using f-strings will mean that any such sequences in the format args will be candidates for expansion.  If `msg` here happens to contain a sequence like `%s`, you'll get a runtime error.  Thus, to be safe, you cannot use f-strings in log messages.

Reply:
Thanks for that explanation, @jwodder. Just tested it out for myself and I see the "not enough format arguments" errors that would result from `%` characters in the first argument.

Reply:
> To correct

for my own sake -- where was I wrong?

> I'm truly not concerned about "wasting cycles"

people without concerns are of the happiest kind.  But some folks need to be concerned and I guess they are the grumpy ones.

> > logging module will always try to interpolate % sequences in log messages
>  @jwodder. Just tested it out for myself and I see the "not enough format arguments" errors

for completeness of this exercise:
- what was exactly you have tested @waxlamp?

```
❯ python3 -c 'import logging; logging.warning("%s is a problem?")'
WARNING:root:%s is a problem?
```
- I guess it is not happening since it is conditional on having those args at https://github.com/python/cpython/blob/3.11/Lib/logging/__init__.py#L376 but may be there is some other location? where do you have that "not enough" spit out at?

Reply:
@yarikoptic The only part of your comment that needed "correction" was "if later decide to mix in adding proper `%-interpolations`".  F-strings in log messages are a problem even if you don't use any `%`-specifiers.

Reply:
Actually, I appear to be wrong, and relatedly, the reason Yarik's attempt didn't work is that ["No % formatting operation is performed on msg when no args are supplied."](https://docs.python.org/3/library/logging.html#logging.Logger.debug)

Reply:
Thanks for this, John. So it appears the idea is: either use `logging.<whatever>()` with only an f-string and no subsequent arguments, or don't use f-strings and feel free to use subsequent arguments. My preference is the former when it's clearer, but clearly both calling sequences have their place.

My own testing ran into that error when I had a format string with, e.g., more than one `%` in it, followed by only one argument. I didn't think to try only a `msg` argument with no following arguments.

Reply:
Given this discussion over f-strings and the `logging` module, I think it would be good to accept Jake's suggestion at the top of this thread, and then @yarikoptic, you can merge this PR.